### PR TITLE
move RangeRegistry to a separate package for generic usage

### LINKED ISSUE #29668: move RangeRegistry to a separate package
FYI. I am implementing other resource allocation for our internal kubernetes version, and want to use this interface, now it resides in `service` package, but I really want to move it to other package like `rangeallocation`. @smarterclayton @lavalamp wdyt?

### PR DESCRIPTION
**Pull request by adohe-zz**
Fix #29668, @smarterclayton @lavalamp ptal.

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/29722)

### CODE REVIEW

**Comment by AdoHe** [comment]
@lavalamp kindly ping.

File: pkg/registry/rangeallocation/registry.go [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package rangeallocation provides the Registry interface and its RESTStorage
+// implementation for storing RangeAllocation api objects.
```
Comment (lavalamp) [inline pkg/registry/rangeallocation/registry.go]: Don't repeat the doc comment

**Comment by lavalamp** [comment]
one nit

**Comment by AdoHe** [comment]
@lavalamp I update this, ptal.

File: pkg/registry/rangeallocation/registry.go (hunk 1 above)
Comment (AdoHe) [inline pkg/registry/rangeallocation/registry.go]: @lavalamp I am afraid if I remove the header here, some pre-commit hook will fail.

File: pkg/registry/rangeallocation/doc.go [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package rangeallocation provides the Registry interface for storing RangeAllocation
+// api objects.
+package rangeallocation // import "k8s.io/kubernetes/pkg/registry/rangeallocation"
```
Comment (smarterclayton) [inline pkg/registry/rangeallocation/doc.go]: Remove this import comment

**Comment by smarterclayton** [comment]
One last comment then I'll apply the label

**Comment by AdoHe** [comment]
@k8s-bot test this issue: #IGNORE

**Comment by AdoHe** [comment]
@smarterclayton please help add `LGTM` label, thanks.

**Pull request merged**